src/test/scala/ContextTests.scala

```diff
@@ -0,0 +1,57 @@
+import org.scalatest.flatspec.AnyFlatSpec
+
+class ContextTests extends AnyFlatSpec:
+  val engine = EngineMock(new IO {}, StorageMock())
+  val gameObject = GameObjectMock(true)
+  def context: Context = Context(engine, gameObject)
+
+  "Context" should "provide a direct reference to the engine IO" in:
+    context.io == engine.io
+
+  it should "provide a direct reference to the engine Storage" in:
+    context.storage == engine.storage
+
+  it should "provide a direct reference to the engine deltaTimeNanos" in:
+    context.deltaTimeNanos == engine.deltaTimeNanos
+
+  it should "provide utility for converting deltaTimeNanos into seconds" in:
+    context.deltaTimeSeconds == 1
```

Comment:
Magari metterei engine.deltaTimeNanos / Mat.pow(10,9) al posto di 1, tanto hai già testato che engine.deltaTimeNanos e context.deltaTimeNanos sono uguali

Reply:
Il motivo per cui ho fatto così è perché mettere `engine.deltaTimeNanos / Mat.pow(10,9) ` significherebbe andare a duplicare il codice di produzione dentro al test. In questo caso sembra banale ma se porti lo stesso ragionamento all'estremo significherebbe dover pensare l'implementazione prima di scrivere il test.

Credi che dovrei comunque cambiare?

Reply:
Ma così dovresti mettere che context.deltaTimeSeconds è uguale a 1 quando lo chiami, e dover fare un altro test cambiando context.deltaTimeNanos e poi uguagliarlo a qualcos'altro, così da avere due test con due context.deltaTimeNanos diversi che controllano context.deltaTimeSeconds e in implementazione metti la parte in comune (ovvero la Mat.pow()) (sarebbe la cosa più giusta da fare)